src/core/Microsoft.PowerToys.Settings.UI.Lib/Utilities/SystemIOProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+
+namespace Microsoft.PowerToys.Settings.UI.Lib.Utilities
+{
+    public class SystemIOProvider : IIOProvider
+    {
+        public bool CreateDirectory(string path)
```

Comment:
[nit] I noticed the return value isn't used anywhere right now. Has it been added since it could be used in the future?

Reply:
Yeah I put it in to give us the option to check. The current code base doesn't check so it will likely throw an exception on  the WriteAllText.  We can probably be more defensive about this (Potentially in alekhya's refactoring).  For now I was trying to minimize changes to logic, and we can improve the implementation seperately.